xarray/backends/common.py

```diff
@@ -503,3 +503,43 @@ def assert_open(self):
         if not self._isopen:
             raise AssertionError('internal failure: file must be open '
                                  'if `autoclose=True` is used.')
+
+
+class PickleByReconstructionWrapper(object):
```

Comment:
It would be nice to add a unit test verifies that this works properly independently of any concrete datastore.

Maybe something simple with `open()`?

Reply:
Done in 8b286c0